articles/tutorials/advanced/2d_shaders/07_sprite_vertex_effect/index.md

```diff
@@ -0,0 +1,863 @@
+	var elements = new VertexElement[] 
+	{ 
+		new VertexElement(0, VertexElementFormat.Vector3, VertexElementUsage.Position, 0), 
+		new VertexElement(12, VertexElementFormat.Color, VertexElementUsage.Color, 0), 
+		new VertexElement(16, VertexElementFormat.Vector2, VertexElementUsage.TextureCoordinate, 0) 
+	};
+	VertexDeclaration = new VertexDeclaration(elements);
+}
+```
+
+> [!note]
+> MonoGame is open source, so you can go read the full code for [SpriteBatchItem](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/SpriteBatchItem.cs) and[`VertexPositionColorTexture`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Vertices/VertexPositionColorTexture.cs)
+
+
+The vertex shader declares a semantic for each input using the `:` syntax:
+
+```hlsl
+// the POSITION0 is the semantic 
+float4 position	: POSITION0,
+```
+
+>[!warning] 
+> You cannot change the `SpriteBatch` vertex shader. 
+> 
+> The `SpriteBatch` class does not offer any way to change the vertex semantics that are passed to the shader function.   
+
+### Output Semantics
+
+The same concept of semantics applies to the output of the shader. Here is the output type of the vertex shader function. Notice that the fields also have the `:` semantic syntax. These semantics instruct the graphics pipeline how to use the data:
+
+```hlsl
+struct VSOutput
+{
+	float4 position		: SV_Position;
+	float4 color		: COLOR0;
+    float2 texCoord		: TEXCOORD0;
+};
+```
+
+This is the _input_ struct for the standard pixel shaders from previous chapters. Notice how the fields are named slightly differently, but the _semantics_ are identical:
+
+```hlsl
+struct VertexShaderOutput  
+{  
+   float4 Position : SV_POSITION;  
+   float4 Color : COLOR0;  
+   float2 TextureCoordinates : TEXCOORD0;  
+};
+```
+
+### Matrix Transform
+
+The default sprite vertex shader uses this line:
+
+```hlsl
+output.position = mul(position, MatrixTransform);
+```
+
+The reason this line exists is to convert the vertices from world-space to clip-space. A vertex is a 3d coordinate in "world-space". But a monitor is a 2d display. Often, the screen's 2d coordinate system is called "clip-space". The vertex shader is converting the 3d world-space coordinate into a 2d clip-space coordinate. That conversion is a vector and matrix multiplication, using the `MatrixTransform`. 
+
+The `MatrixTransform` is computed by the [`SpriteEffect`](xref:Microsoft.Xna.Framework.Graphics.SpriteEffect) class. The full source is available, [`here`](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/Effect/SpriteEffect.cs#L63). The relevant lines are copied below:
+
+```csharp
+// cache the shader parameter for the MatrixTransform
+_matrixParam = Parameters["MatrixTransform"];
+
+// ... some code left out for readability
+
+// create a projection matrix in the _projection variable
+Matrix.CreateOrthographicOffCenter(0, vp.Width, vp.Height, 0, 0, -1, out _projection);
+
```

Comment:
Include links for each type to a common reference article, such as https://docs.monogame.net/articles/getting_to_know/howto/graphics/camera/HowTo_Understand_Different_Camera_Modes.html 

To better explain the concepts to users.  2D readers might struggule with perspective (pun intended) over orthographic which they might even realise they are using.